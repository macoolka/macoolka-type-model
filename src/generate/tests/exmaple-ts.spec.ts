import * as E from 'fp-ts/lib/Either'
import { pipe } from 'fp-ts/lib/pipeable'
import { writeFileSync } from 'fs'
import {InputModule, ioBuild, typeBuild } from '../../'
const userSchema: InputModule.MModule = {
    name: 'User',
    interfaces: [{
        name: 'User',
        fields: [{
            name: 'name',
            required: true,
        }, {
            name: 'id',
            id: true,
            required: true,
        }, {
            name: 'age',
            type: 'int',
        }, {
            name: 'female',
            type: 'boolean'
        }, {
            name: 'city',
            type: {
                _kind: 'enum',
                values: ['dalian', 'london', 'newyork', 'beijing'],
                defaultValue: 'dalian',
            },
            required: true,
        }]
    }]
}

describe('example', () => {
    it('build input type', () => {
        pipe(
            userSchema,
            typeBuild({ isInput: true  }),
            result => {
                expect(E.isRight(result)).toBeTruthy()
                return result;
            },
            E.map(content => {
                expect(content).toMatchSnapshot()
                writeFileSync(__dirname + '/UserInput.ts', content)
            })
        )
    })
    it('build type', () => {
        pipe(
            userSchema,
            typeBuild(),
            result => {
                expect(E.isRight(result)).toBeTruthy()
                return result;
            },
            E.map(content => {
                expect(content).toMatchSnapshot()
                writeFileSync(__dirname + '/User.ts', content)
            })
        )

    })
    it('build valid io', () => {
        pipe(
            userSchema,
            ioBuild(),
            result => {
                expect(E.isRight(result)).toBeTruthy()
                return result;
            },
            E.map(content => {
                expect(content).toMatchSnapshot()
                writeFileSync(__dirname + '/User.io.ts', content)
            })
        )

    })

})



