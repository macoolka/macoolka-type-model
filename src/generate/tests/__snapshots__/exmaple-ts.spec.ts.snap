// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example build input type 1`] = `
"/**
 * Code generated by github.com/macoolka/macoolka-gen-model, DO NOT EDIT.
 */
export interface NonEmptyArray<A> extends Array<A> {
    0: A
}

export type MaybeArray<A> = A | Array<A>

export interface User {
  name: string
  id: string
  age?: number
  female?: boolean
  city?: 'dalian' | 'london' | 'newyork' | 'beijing'
}


"
`;

exports[`example build type 1`] = `
"/**
 * Code generated by github.com/macoolka/macoolka-gen-model, DO NOT EDIT.
 */
export interface NonEmptyArray<A> extends Array<A> {
    0: A
}

export type MaybeArray<A> = A | Array<A>

export interface User {
  name: string
  id: string
  age?: number
  female?: boolean
  city: 'dalian' | 'london' | 'newyork' | 'beijing'
}


"
`;

exports[`example build valid io 1`] = `
"/**
 * Code generated by github.com/macoolka/macoolka-gen-model, DO NOT EDIT.
 */
import * as t from 'macoolka-io'
export const User =t.intersection([  t.type({
    name:t.string,
    id:t.string,
    city:t.withDefault(t.keyof({\\"dalian\\":\\"\\",\\"london\\":\\"\\",\\"newyork\\":\\"\\",\\"beijing\\":\\"\\"}),\\"dalian\\"),
  })
,  t.partial({
    age:t.int,
    female:t.boolean,
  })
])



"
`;
